src/coap_client.c

```diff
@@ -261,6 +261,34 @@ static enum golioth_status golioth_coap_client_set_internal(struct golioth_clien
     return GOLIOTH_OK;
 }
 
+enum golioth_status golioth_coap_client_post(struct golioth_client *client,
```

Comment:
Do we have anywhere else that we use `_post` in the user API? I ask because this is a departure from the `_set` and `_set_block` API naming we currently have. I wonder if `_set_payload` or something along those lines would be a good fit?

Reply:
Yeah... I was thinking about that myself and considered `set_payload` / `set_with_response` / `set_with_rsp`, etc.

1. My first implementation was about extending `set` with payload (and payload_size) arguments. This seemed to be obvious choice for internals of our SDK and was the cleanest path forward, without introducing new API (like _post). This turned out to be very verbose for public API that SDK users consume. So... decided to leave those APIs **without** payload.
2. Next approach was to implement either `_set()` equivalent with `_set_with_rsp()` to describe that payload is provided in that case. That however was very loosely related to our existing `_set()` APIs, like `lightdb_*_set()`. Then I was thinking about whether `_set_*()` is a good choice for API that sends and receives something. More natural is the `_post()` nomenclature, which is very well known in case of HTTP, CoAP and in general in the world of RESTful APIs.
3. Decided to go with `_post()` API, since that maps very good to internal CoAP implementation without introducing new nomenclature. It does what POST method does technically... so it sends data and receives data. It is definitely more than **set** word describes.

So if we are about changing the (internal) APIs, my bet would be to use `_post()` in most places and just expose `_set()` for high-level APIs, like lightdb, where it makes sense. Even for **stream** service which just sends data now... I think it makes more sense to use `golioth_stream__push()` or `golioth_stream__post()` instead of `golioth_stream__set()`.

Reply:
I also like using `_post` because it more closely matches our use of `_get` and `_delete` elsewhere. I don't love the idea of us now exposing a `set_cb` and a `post_cb` as part of the public API though. The `post_cb` looks very similar to the `get_cb` (identical?) and the `set_cb` is currently used for POST and DELETE operations. I wonder if moving to something more like "callback with response" and "callback without response" would make more sense?